server_test.go

```diff
@@ -344,3 +344,18 @@ func TestOpenStatRace(t *testing.T) {
 	os.Remove(tmppath)
 	checkServerAllocator(t, server)
 }
+
+// Ensure that proper error codes are returned for non existent files, such
+// that they are mapped back to a 'not exists' error on the client side.
+func TestStatNonExistent(t *testing.T) {
+	client, server := clientServerPair(t)
+	defer client.Close()
+	defer server.Close()
+
+	for _, file := range []string{"/doesnotexist", "/doesnotexist/a/b"} {
+		_, err := client.Stat(file)
+		if err == nil || !os.IsNotExist(err) {
```

Comment:
`os.IsNotExist(nil)` will already return false, so we don‘t really need the `err == nil` check. (While not explicitly documented behavior, this should follow logically from the contract of the function, that is, since `err == nil` is not a file not found error, and should therefore never return true.)

https://play.golang.org/p/lFB2yI3oQ5i

Reply:
done